### PR TITLE
Sleep-generate cycles

### PR DESCRIPTION
**Pull request by hariso**
### Description

Please include a summary of the change and what type of change it is (new feature, bug fix, refactoring, documentation).
Please also include relevant motivation and context.
List any dependencies that are required for this change.

Fixes # (issue)

### Quick checks:

- [ ] I have followed the [Code Guidelines](https://github.com/ConduitIO/conduit/blob/main/docs/code_guidelines.md).
- [ ] There is no other [pull request](https://github.com/ConduitIO/conduit-connector-generator/pulls) for the same update/change.
- [ ] I have written unit tests.
- [ ] I have made sure that the PR is of reasonable size and can be easily reviewed.

### CODE REVIEW

File: source_test.go [hunk 1]
```diff
@@ -88,3 +88,66 @@ func TestRead_StructuredData(t *testing.T) {
...
+		results <- result{err: err, duration: time.Since(start)}
+	}()
+
+	select {
+	case r := <-results:
+		is.NoErr(r.err)
+		is.True(r.duration >= 40*time.Millisecond) // expected source to sleep for given time
+	case <-time.After(50 * time.Millisecond):
+		is.Fail() // timed out waiting for record
+	}
```
Comment (lovromazgon) [inline source_test.go:152]: Can we add another read after this select? That one should then again happen after ~210ms, right?

File: README.md [hunk 2]
```diff
@@ -22,9 +22,11 @@ The connector is great for getting started with Conduit but also for certain typ
...
-| readTime      | The time it takes to 'read' a record.                                                      | false    | "0s"    |
-| fields        | A comma-separated list of name:type tokens, where type can be: int, string, time, bool.    | true     | ""      |
-| format        | Format of the generated payload data: raw, structured.                                     | false    | "raw"   |
+| name         | description                                                                                | required | Default             |
+|--------------|--------------------------------------------------------------------------------------------|----------|---------------------|
+| recordCount  | Number of records to be generated. -1 for no limit.                                        | false    | "-1"                |
+| readTime     | The time it takes to 'read' a record.                                                      | false    | "0s"                |
+| sleepTime    | The time the generator 'sleeps' before it starts generating records. Must be non-negative. | false    | "0s"                |
```
Comment (lovromazgon) [inline README.md:29]: The description sounds to me like the field controls how long the generator waits before starting to generate records _the first time_, while in reality the sleep time will be applied repeatedly after `generateTime` is reached. I think we should clarify this.

File: README.md [hunk 3]
```diff
@@ -22,9 +22,11 @@ The connector is great for getting started with Conduit but also for certain typ
...
+| generateTime | The amount of time the generator is generating records. Must be positive.                  | false    | max. duration in Go |
```
Comment (lovromazgon) [inline README.md:30]: `generateTime` won't make a difference unless `sleepTime` is specified as well. This should also be noted.

Comment (lovromazgon) [inline README.md:30]: Given that these two options belong together and their purpose is to create bursts we could group them together as `burst.sleepTime` and `burst.generateTime` (or something along this way).

File: source_test.go (hunk 1 above)
Comment (hariso) [reply source_test.go:152]: Makes sense, I just pushed it.

File: README.md (hunk 3 above)
Comment (hariso) [reply README.md]: Yup, I think that would make it better. Pushed the changes just now.

Comment (hariso) [reply README.md]: Good one, just pushed the changes.

File: README.md [hunk 4]
```diff
@@ -20,13 +20,17 @@ explained in more details in the [Configuration section](#Configuration) below.
...
-| format       | Format of the generated payload data: raw, structured.                                     | false    | "raw"               |
+| name               | description                                                                                                          | required | Default             |
+|--------------------|----------------------------------------------------------------------------------------------------------------------|----------|---------------------|
+| recordCount        | Number of records to be generated. -1 for no limit.                                                                  | false    | "-1"                |
+| readTime           | The time it takes to 'read' a record.                                                                                | false    | "0s"                |
+| burst.sleepTime    | The time the generator 'sleeps'. Must be non-negative.                                                               | false    | "0s"                |
+| burst.generateTime | The amount of time the generator is generating records. Must be positive.<br/>Has effect only if `sleepTime` is set. | false    | max. duration in Go |
```
Comment (lovromazgon) [inline README.md:34]:
```suggestion
| burst.generateTime | The amount of time the generator is generating records. Must be positive.<br/>Has effect only if `burst.sleepTime` is set. | false    | max. duration in Go |
```

File: README.md [hunk 5]
```diff
@@ -20,13 +20,17 @@ explained in more details in the [Configuration section](#Configuration) below.
...
-| fields       | A comma-separated list of name:type tokens, where type can be: int, string, time, bool.    | true     | ""                  |
-| format       | Format of the generated payload data: raw, structured.                                     | false    | "raw"               |
+| name               | description                                                                                                          | required | Default             |
+|--------------------|----------------------------------------------------------------------------------------------------------------------|----------|---------------------|
+| recordCount        | Number of records to be generated. -1 for no limit.                                                                  | false    | "-1"                |
+| readTime           | The time it takes to 'read' a record.                                                                                | false    | "0s"                |
+| burst.sleepTime    | The time the generator 'sleeps'. Must be non-negative.                                                               | false    | "0s"                |
```
Comment (lovromazgon) [inline README.md:33]:
```suggestion
| burst.sleepTime    | The time the generator 'sleeps' between bursts. Must be non-negative.                                                | false    | "0s"                |
```

**Review by lovromazgon — APPROVED** [review]
Only two suggestions to clarify docs, LGTM all in all 👍

File: README.md (hunk 4 above)
Comment (hariso) [reply README.md:34]: Good one. :+1:

File: README.md (hunk 5 above)
Comment (hariso) [reply README.md]: Also a good one. :+1:

**Pull request merged**